Rubberduck.Parsing/Symbols/DeclarationFinder.cs

```diff
@@ -778,9 +777,22 @@ private bool IsProjectGlobalDeclaration(Declaration candidateDeclaration, Declar
         {
             return candidateDeclaration.ProjectName == scopingDeclaration.ProjectName
                 && !(candidateDeclaration.ParentScopeDeclaration is ClassModuleDeclaration)
-                && (candidateDeclaration.Accessibility == Accessibility.Public
-                    || ((candidateDeclaration.Accessibility == Accessibility.Implicit)
-                        && (candidateDeclaration.ParentScopeDeclaration is ProceduralModuleDeclaration)));
+                && (IsExplicitPublicInOtherModule(candidateDeclaration, scopingDeclaration)
+                || IsImplicitPublicInOtherModule(candidateDeclaration, scopingDeclaration));
+        }
+
+        private bool IsExplicitPublicInOtherModule(Declaration candidateDeclaration, Declaration scopingDeclaration)
+        {
+            return candidateDeclaration.ComponentName != scopingDeclaration.ComponentName
+                    && candidateDeclaration.Accessibility == Accessibility.Public;
+        }
+
+        private bool IsImplicitPublicInOtherModule(Declaration candidateDeclaration, Declaration scopingDeclaration)
+        {
+            return candidateDeclaration.ComponentName != scopingDeclaration.ComponentName
+                        && candidateDeclaration.Accessibility == Accessibility.Implicit
+                        && (candidateDeclaration.ParentScopeDeclaration is ProceduralModuleDeclaration)
+                        && !candidateDeclaration.IdentifierName.StartsWith("Option ");
```

Comment:
By "this" I'm guessing it is the StartsWith(..) call?  I could find no reliable parse tree content to detect/compare module options.  Without the check, a module option in module A shows up as a declaration 'visible' to code in module/class B.  So, the string operation is where I landed.  As for the explicit/implicit Public - I'm happy to combine the functions.  I generally look at how the 'top-level' function looks and prefer to have it read like a description so it is clear what's being checked (especially when I come back to it in a couple months).  To my less experienced C# eyes (no pun intended), the LINQ syntax gets in the way of clarity (for me).  I think you are right that in this case, it is probably overkill.